ui/src/route.rs

```diff
@@ -16,22 +16,33 @@ pub enum Route {
 ////////////////////////////////////////////////////////////////
 
 impl Route {
-    pub fn as_str(self) -> &'static str {
+    pub fn path(&self) -> Vec<&str> {
         match self {
-            Route::Title => "",
-            Route::StartGame => "start-game",
-            Route::Demo => "demo",
+            Route::Title => vec![],
+            Route::StartGame => vec!["start-game"],
+            Route::Demo => vec!["demo"],
         }
     }
+    pub fn to_string(self) -> String {
+        let mut buf = String::new();
+
+        buf.push('/');
+
+        buf.push_str(&self.path().join("/"));
```

Comment:
`buf.push('/')` ensures the url starts with a `/`, which turns out to be really critical to proper routing.
```
<a href="start-game"/>
```
reloads the whole page and restarts all the code
```
<a href="/start-game"/>
```
does not